### PR TITLE
Added custom PutEvent to Batch Starter Lambda

### PR DESCRIPTION
**Pull request by tech3371**
# Change Summary
Added custom PutEvent to batch starter lambda which sends event to EventBridge. And There is EventBridge rule on indexer lambda that listens for this event.

## New Files
- sds_data_manager/lambda_code/SDSCode/lambda_custom_events.py
   - Dataclass to store EventBridge event details
 

## Updated Files
- sds_data_manager/lambda_code/batch_starter.py
   - Added function to send custom event to EventBridge from lambda
- tests/lambda_endpoints/conftest.py
   - mock events

## Testing
- tests/lambda_endpoints/test_batch_starter.py

### CODE REVIEW

File: sds_data_manager/lambda_code/batch_starter.py [hunk 1]
```diff
@@ -338,6 +336,60 @@ def extract_components(filename: str):
...
+    command = command_parameters.split("--")
+    s3_uri = command[3].replace("s3_uri '", "").replace("' ", "")
+    dependency = command[-1].replace("dependency ", "")
+
+    # Create event["detail"] information
+    detail = EventDetail(
+        file_to_create=s3_uri,
+        status=models.Status.INPROGRESS.value,
+        dependency=dependency,
+    )
```
Comment (greglucas) [inline sds_data_manager/lambda_code/batch_starter.py]: I will just voice my slight preference for not using dataclasses everywhere and leaving this as a simple dictionary and avoiding the `asdict(EventDetail)` call in the other dataclass. I know I'm in the minority here though, so it is fine to keep it as-is too.
```suggestion
    detail = {
        "file_to_create": s3_uri,
        "status": models.Status.INPROGRESS.value,
        "dependency": dependency,
    }
```

**Review by greglucas — APPROVED** [review]

File: sds_data_manager/lambda_code/batch_starter.py [hunk 2]
```diff
@@ -338,6 +336,60 @@ def extract_components(filename: str):
     return components
 
 
+def send_lambda_put_event(command_parameters):
+    """Sends custom PutEvent to EventBridge
```
Comment (laspsandoval) [inline sds_data_manager/lambda_code/batch_starter.py]: So glad you have an example here. My question is: I thought that this was the information that would be used to populate the StatusTracking table, which has the following columns. 

    id = Column(Integer, Identity(start=1, increment=1), primary_key=True)
    file_to_create_path = Column(String, nullable=False)
    status = Column(statuses, nullable=False)
    job_definition = Column(String, nullable=False)
    ingestion_date = Column(DateTime, nullable=True)
    
We've had a lot of conversations so maybe that's changed. You might just need to get me up to speed on this.

**Comment by laspsandoval** [comment]
Just a comment on the PUT event. We can discuss at the meeting. Looks good besides that.

File: sds_data_manager/lambda_code/batch_starter.py (hunk 2 above)
Comment (tech3371) [reply sds_data_manager/lambda_code/batch_starter.py]: That's true. This event is suppose to send custom putevent from batch starter to indexer lambda to populate status table. But we send this information to initialized that table with `file_to_create`, `status` and initialize others a null. Then when batch job finishes we update table with new information and update status as well based on batch job event.

**Comment by tech3371** [comment]
pre-commit.ci autofix

**Pull request merged**